### PR TITLE
Refactor Auto Review

### PR DESCRIPTION
**Pull request by nateshim-indico**

### CODE REVIEW

File: examples/auto_review_predictions.py [hunk 1]
```diff
@@ -26,17 +30,12 @@
...
-        {
-            "function": "accept_by_confidence",
-            "kwargs": {"conf_threshold": 0.98, "labels": ["Name", "Amount"]},
-        },
-    ]
-)
-auto_reviewer = AutoReviewer(predictions, review_config)
+# Set up review functions and review predictions
+functions = [
+    AutoReviewFunction(remove_by_confidence, kwargs={"conf_threshold": 0.90}),
```
Comment (nateshim-indico) [inline examples/auto_review_predictions.py:35]: default to all labels; also call that out in docstring in examples and auto review file

Comment (nateshim-indico) [reply examples/auto_review_predictions.py]: resolved

File: indico_toolkit/auto_review/auto_reviewer.py [hunk 2]
```diff
@@ -1,25 +1,30 @@
...
+        self,
+        function: Callable,
+        labels: List[str] = [],
+        kwargs: Dict[str, str] = {},
+    ):
+        self.function = function
+        self.labels = labels
+        self.kwargs = kwargs
 
+    def apply(self, predictions: List[dict] = []):
```
Comment (Scott771) [inline indico_toolkit/auto_review/auto_reviewer.py:23]: mutable default arguments (list) can have issues, do we need a default argument here?

File: examples/auto_review_predictions.py [hunk 3]
```diff
@@ -26,17 +29,12 @@
...
-        {"function": "remove_by_confidence", "kwargs": {"conf_threshold": 0.90}},
-        {
-            "function": "accept_by_confidence",
-            "kwargs": {"conf_threshold": 0.98, "labels": ["Name", "Amount"]},
-        },
-    ]
-)
-auto_reviewer = AutoReviewer(predictions, review_config)
+# Set up review functions and review predictions
+functions = [
```
[the remark this answers is not in the document]
Comment (nateshim-indico) [reply examples/auto_review_predictions.py:41]: added `custom_function` in example

File: indico_toolkit/auto_review/auto_reviewer.py [hunk 4]
```diff
@@ -1,25 +1,30 @@
...
-    accept_by_confidence,
-    reject_by_confidence,
-    reject_by_min_character_length,
-    reject_by_max_character_length,
-    accept_by_all_match_and_confidence,
-    remove_by_confidence,
-)
+class AutoReviewFunction:
+    """
+    Class for hosting functions to manipulate predictions before sending to
```
[the remark this answers is not in the document]
Comment (nateshim-indico) [reply indico_toolkit/auto_review/auto_reviewer.py:5]: called out needed format for Callable in docstring

File: indico_toolkit/auto_review/auto_reviewer.py (hunk 2 above)
Comment (nateshim-indico) [reply indico_toolkit/auto_review/auto_reviewer.py]: removed default arg

**Review by Scott771 — APPROVED** [review]

**Pull request merged**